src/System.Linq/src/System/Linq/Select.cs

```diff
@@ -170,9 +172,58 @@ public int GetCount(bool onlyIfCheap)
             {
                 return _source.Length;
             }
+
+            public IPartition<TResult> Skip(int count)
+            {
+                if (count == 0) return (IPartition<TResult>)Clone();
+                if (count >= _source.Length) return new EmptyPartition<TResult>();
+                return new SelectIPartitionIterator<TSource, TResult>(new ListPartition<TSource>(_source, count, int.MaxValue), _selector);
+            }
+
+            public IPartition<TResult> Take(int count)
+            {
+                return count >= _source.Length ? (IPartition<TResult>)Clone() : new SelectIPartitionIterator<TSource, TResult>(new ListPartition<TSource>(_source, 0, count - 1), _selector);
+            }
```

Comment:
> can be enumerated through in half the time. The savings can be very large with some queries.

Wouldn't it be better to instead write that as:
```C#
arrayOfThousandElements.Skip(250).Take(250).Select(i => i.ToString())
```
?

In general, I'm not too concerned with saving a developer from themselves; with all of these LINQ optimizations, I'd like the focus to be on minimizing the overhead for well-written queries, rather than trying to fix poorly written ones (if the benefits carry over to the poor ones, that's good, too).